packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -153,32 +153,47 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
   const toggleButtonRef = React.useRef<HTMLSpanElement>(null);
   const mounted = React.useRef(false);
 
+  // Record to store all extra information (e.g. original indexes), where the key is the id of the option
+  const optionsExtraInfoRef = React.useRef<
+    Record<string, { __originalIndex: number }>
+  >({});
+
+  // Function to initialize extra info
+  const initializeExtraInfo = React.useCallback(() => {
+    optionsRef.current.forEach((option, index) => {
+      optionsExtraInfoRef.current[getOptionId(option, id)] = {
+        __originalIndex: index,
+      };
+    });
+  }, [id]);
+
+  // Latest value of the valueProp prop
+  const valuePropRef = React.useRef(valueProp);
+  valuePropRef.current = valueProp;
+
   // Latest value of the onChange prop
   const onChangeProp = React.useRef(onChange);
   React.useEffect(() => {
     onChangeProp.current = onChange;
   }, [onChange]);
 
-  // Record to store all extra information (e.g. original indexes), where the key is the id of the option
-  const optionsExtraInfoRef = React.useRef<
-    Record<string, { __originalIndex: number }>
-  >({});
+  // Latest value of the options prop
+  const optionsRef = React.useRef(options);
+  optionsRef.current = options;
 
-  // Clear the extra info when the options change so that it can be reinitialized below
+  // Change options ref on options change
   React.useEffect(() => {
+    // Clear the extra info when the options change so that it can be reinitialized below
     optionsExtraInfoRef.current = {};
-  }, [options]);
+    initializeExtraInfo();
```

Comment:
I don't think we need this change. When useEffect clears extra info, it will automatically run the logic below which checks `Object.keys(optionsExtraInfoRef.current).length === 0`, so no extra info needed.

Lets remove this function and bring back the original order

Reply:
Removed and restored original order